pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java

```diff
@@ -80,4 +85,88 @@
   double[] getDoubleArray(int rowId, int colId);
 
   String[] getStringArray(int rowId, int colId);
+
+  /* The MetadataKeys is used in V3, where we present metadata as Map<MetadataKeys, String>
+   * ATTENTION:
+   *  - Don't add new key which has same id/name with existing keys. Duplicate name is not allowed.
+   *  - Don't change name of existing keys.
+   *  - Don't remove existing keys.
+   *  - Always add new keys to the end.
+   *  Otherwise, backward compatibility will be broken.
+   */
+  enum MetadataKeys {
+    UNKNOWN("unknown"),
```

Comment:
Why are we using the constructor based pattern ? The getName function of enum will automatically return the constant name as string in CAMEL case which is exactly what you are passing to the constructor. So we don't need constructor based creation of enum constants. You can simply define the constant and Java enum implementation will take care of the name

Reply:
This is a good suggestion, but we should be careful about the strings that:
- Broker uses in returning metadata
- Broker/Server use in logs.
It is better we don't change those strings.

Reply:
No longer needed based on the comment https://github.com/apache/incubator-pinot/pull/6710/files#r603639040

Reply:
done